NickvisionMoney.Shared/Models/CurrencyConversionService.cs

```diff
@@ -0,0 +1,161 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace NickvisionMoney.Shared.Models;
+
+/// <summary>
+/// A model of a result of a currency conversion
+/// </summary>
+public class CurrencyConversion
+{
+    /// <summary>
+    /// The currency code of the source amount
+    /// </summary>
+    public string SourceCurrnecy { get; init; }
+    /// <summary>
+    /// The source amount to convert
+    /// </summary>
+    public decimal SourceAmount { get; init; }
+    /// <summary>
+    /// The currency code for the result amount
+    /// </summary>
+    public string ResultCurrency { get; init; }
+    /// <summary>
+    /// The rate of conversion from the source currency to the result currency
+    /// </summary>
+    public decimal ConversionRate { get; init; }
+
+    /// <summary>
+    /// The result amount
+    /// </summary>
+    public decimal ResultAmount => SourceAmount / ConversionRate;
+    
+    /// <summary>
+    /// Constructs a CurrencyConversion
+    /// </summary>
+    /// <param name="sourceCurrency">The currency code of the source amount</param>
+    /// <param name="sourceAmount">The source amount to convert</param>
+    /// <param name="resultCurrency">The currency code for the result amount</param>
+    /// <param name="conversionRate">The rate of conversion from the source currency to the result currency</param>
+    public CurrencyConversion(string sourceCurrency, decimal sourceAmount, string resultCurrency, decimal conversionRate)
+    {
+        SourceCurrnecy = sourceCurrency;
+        SourceAmount = sourceAmount;
+        ResultCurrency = resultCurrency;
+        ConversionRate = conversionRate;
+    }
+}
+
+/// <summary>
+/// A service for getting currnecy conversions
+/// </summary>
+public class CurrencyConversionService
+{
+    private static readonly HttpClient _http;
+    
+    /// <summary>
+    /// Constructs a static CurrencyConversionService
+    /// </summary>
+    static CurrencyConversionService()
+    {
+        _http = new HttpClient();
+        _http.DefaultRequestHeaders.UserAgent.ParseAdd("Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/116.0");
+    }
+
+    /// <summary>
+    /// Converts a sourceAmount from the sourceCurrency to the resultCurrency
+    /// </summary>
+    /// <param name="sourceCurrency">The currency code of the source amount</param>
+    /// <param name="sourceAmount">The source amount to convert</param>
+    /// <param name="resultCurrency">The currency code for the result amount</param>
+    /// <returns>CurrencyConversion if successful, else null</returns>
+    public static async Task<CurrencyConversion?> ConvertAsync(string sourceCurrency, decimal sourceAmount, string resultCurrency)
+    {
+        if (sourceCurrency == resultCurrency)
+        {
+            return new CurrencyConversion(sourceCurrency, sourceAmount, resultCurrency, sourceAmount);
```

Comment:
```suggestion
            return new CurrencyConversion(sourceCurrency, sourceAmount, resultCurrency, 1);
```
Conversion rate should be 1 if identical currencies